packages/cloud_firestore_odm/cloud_firestore_odm_generator/lib/src/collection_data.dart

```diff
@@ -336,6 +344,22 @@ represents the content of the collection must be in the same file.
     }
   }
 
+  static void _assertValidJsonAnnotation(
+    DartType type, {
+    required bool hasPerFieldToJson,
+  }) {
+    final isEnumField = type.isEnum;
+    final hasEnumTypeArg = type.isDartCoreIterable &&
+        (type as InterfaceType).typeArguments.any((e) => e.isEnum);
+
+    if ((isEnumField || hasEnumTypeArg) && !hasPerFieldToJson) {
+      throw InvalidGenerationSourceError(
```

Comment:
I like the idea! But this is unlikely to work when the flag is set globally in a `build.yaml`

Because in that case, the annotation has `null` for value

Reply:
To make such an error, we'd likely need https://github.com/dart-lang/build/issues/3557

Reply:
This field is also used to actually generate the code, so if it's an issue here it's also an issue elsewhere

Reply:
Assuming this is set causes issues with the `ManualJson` and `PublicRedirected` test classes